[Tan Ying Ting] iP

ytingtan opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen 
> ([source](https://dansilvestre.com/productivity-quotes/))

An interactive chat bot for you to 
- add
- view
- store 
- delete

your tasks. 😸 

Task types available: 
- [ ] Todo  `todo <name of todo>` 
- [ ] Event `event <name of event> /at <location>` 
- [ ] Deadline `deadline <name of deadline> /by <date in YYYY-MM-DD format>`

This is a project template for a greenfield Java project. It's named after the Java mascot _Duke_. Given below are instructions on how to use it.

## Option 1: Setting up in Intellij

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
1. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
   ```
   Hello! I'm duke.Duke
   What can I do for you?
   ```
   
## Option 2: Using JAR file 

1. Copy `ip.jar` file into an empty folder 
2. Open a command window in that folder 
3. Run the command `java -jar ip.jar` (i.e., run the command in the same folder as the jar file).
4. If setup is correct, you should see below as the output: 
   ```
   Hello! I'm duke.Duke
   What can I do for you?
   ```

Commit:
Add Gradle support

ytingtan committed:
Level 1

ytingtan committed:
Ignore .class files

ytingtan committed:
Stores tasks and prints task list

ytingtan committed:
Update printList to use global variable

ytingtan committed:
Made Task class, corrected minor indexing error

ytingtan committed:
Created Todos, Deadlines, Events

ytingtan committed:
Automated Text UI Testing

ytingtan committed:
Added DukeException

ytingtan committed:
Changed data strucutre to ArrayList and added Delete function

ytingtan committed:
Added storage function

ytingtan committed:
Store deadline dates as a java.time.LocalDate

ytingtan committed:
Merge branch-Level-8 into master branch

ytingtan committed:
Add DateTimeParseException to read() method

ytingtan committed:
Merge branch 'add-gradle-support'

xav168 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,181 @@
+        read();
+        while (sc.hasNext()) {
+            String input = sc.nextLine();
+            String[] breakitdown = input.split(" ");
```
Perhaps consider a better name for the string[] also the naming should be in camel case

xav168 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,181 @@
+                    if (!desc.contains("/at")) {
+                        throw new DukeException("OOPS!!! /at keyword not found!");
+                    }
+                    String[] taskNameLocation = desc.split("/at ");
```
could consider better naming for this String[] as well

xav168 reviewed (commented):
LGTM, some naming nits and could consider to add modularity to your code!

ytingtan committed:
Run gradle and resolve errors

ytingtan committed:
Refactor code

Marcusgwj commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
Should this be removed? I noticed the same issue in several other places too.

Marcusgwj commented on src/main/java/duke/Task.java:
```diff
@@ -1,4 +1,4 @@
-package main.java;
+package duke;
 
 
```
Should the extra lines from lines 35 to 38 be removed?

Marcusgwj reviewed (commented):
LGTM, just a few nits to fix! Comments can also be improved when working on Javadocs

ytingtan committed:
Add TaskList class to make code more object-oriented

ytingtan committed:
Add Storage class to make code mroe object-oriented

ytingtan committed:
Add Storage class to make code more object-oriented

ytingtan committed:
Merge branch 'master' of https://github.com/ytingtan/ip

ytingtan committed:
Add UI class and fix minor bugs

Encapsulate System.out.println() commands into a UI class to create clearer division of instructions.

ytingtan committed:
Add Parser class and fix bugs

ytingtan committed:
Add JUnit Tests

Add test cases to test toString methods() of ToDos and Deadlines classes.

ytingtan committed:
Update README file to provide instructions to run Duke as JAR file

ytingtan committed:
Add JavaDoc comments

ytingtan committed:
Tweak the code to comply with a coding standard

ytingtan committed:
Add Find command.
Give users a way to find a task by searching for a keyword.

ytingtan committed:
Add Find command
Give users a way to find a task by searching for a keyword.

ytingtan committed:
Merge branch 'branch-Level-9' of https://github.com/ytingtan/ip into branch-Level-9

ytingtan committed:
Merge branch branch-A-CodingStandard

ytingtan committed:
Merge branch 'branch-Level-9'

ytingtan committed:
Fix minor bugs in find command

Previously, the printOnFind command under the Ui class only printed the first entry that has the keyword.

Now, changed command to print all entries that have the keyword.

ytingtan committed:
Add GUI to Duke

Refactor code into Command package. Use javafx for GUI.

ytingtan committed:
Add GUI

ytingtan committed:
Merge pull request #2 from ytingtan/branch-Level-10

Merge Branch level 10

ytingtan committed:
Fix OutOfBoundsIndex error in Delete function

ytingtan committed:
User Assertions

Previously, the code does not have any assertions.

Assertions are used to document important assumptions that should hold at various points in the code.

Add assertions to check that input is not null.

ytingtan committed:
Refactor Storage.java to maximise code quality

There was major deep nesting (arrowhead style code) in the code.

Refactor code to remove arrowhead style code.

ytingtan committed:
Refactor Parser.java to maximise code quality

Imported the entire package before.

Let's make explicit imports instead.

ytingtan committed:
Refactor Ui.java to maximise code quality

Deleted arrowhead style code to maximise readability.

ytingtan committed:
Minor fixes throughout entire code repository

- Change breakitdown to splitInput (better variable naming)
- Remove unused imports

ytingtan committed:
Merge pull request #3 from ytingtan/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

ytingtan committed:
Resolve merge conflicts

ytingtan committed:
Merge pull request #4 from ytingtan/branch-A-Assertions

User Assertions

ytingtan committed:
Add Help function to Duke

ytingtan committed:
Merge pull request #5 from ytingtan/branch-C-Help

Add Help function to Duke

ytingtan committed:
Add a User Guide

ytingtan committed:
Fix minor error in README.md

ytingtan committed:
Release

ytingtan committed:
Merge branch 'master' of https://github.com/ytingtan/ip
